src/Config/Module.php

```diff
@@ -36,22 +36,40 @@ public function __construct(ConfigInterface $config, ShellInterface $shell)
 
     /**
      * Reconciling installed modules with shared config.
+     * Returns list of enabled modules or an empty array if no modules were enabled.
      *
      * @throws ShellException
      * @throws FileSystemException
      */
-    public function refresh()
+    public function refresh(): array
     {
         $moduleConfig = (array)$this->config->get('modules');
 
+        $process = $this->shell->execute('php ./bin/magento module:enable --all --ansi --no-interaction');
+
         if (!$moduleConfig) {
-            $this->shell->execute('php ./bin/magento module:enable --all --ansi --no-interaction');
             $this->config->reset();
+        } else {
+            $this->config->update(['modules' => $moduleConfig]);
+        }
 
-            return;
+        return $this->parseModuleList($process->getOutput());
```

Comment:
This is not working as expected.  This shows modules that were previously disabled as being enabled, though they remain disabled after.  Steps to recreate:

1. `./vendor/bin/ece-tools module:refresh`
2. Modify `app/etc/config.php`, disable `Temando_Shipping` and `Vertex_Tax` by setting to `0`
3. `./vendor/bin/ece-tools module:refresh`

Expected result:
`No modules were changed.`

Actual result:
```INFO: The following modules have been enabled:
Temando_Shipping
Vertex_Tax
```
Modules are disabled in `app/etc/config.php` still

Reply:
Thanks for your comment. Implementation was fixed.